File: backend/domain/src/main/kotlin/com/dopashift/domain/repository/DailyTodoRepository.kt
package com.dopashift.domain.repository

import com.dopashift.domain.entity.DailyTodoItem
import java.time.LocalDate
import java.util.UUID

/**
 * Port interface for daily to-do item persistence.
 * Implementations live in the infrastructure layer.
 */
interface DailyTodoRepository {
    suspend fun findById(id: UUID): DailyTodoItem?
    suspend fun findByUserIdAndDate(userId: UUID, date: LocalDate): List<DailyTodoItem>
    suspend fun countByUserIdAndDate(userId: UUID, date: LocalDate): Int
    suspend fun save(item: DailyTodoItem): DailyTodoItem
    suspend fun delete(id: UUID)
}

File: web/src/domain/interfaces/DailyTodoRepository.ts
import type { DailyTodo } from '../entities/DailyTodo';

/** Port interface for daily to-do item persistence. */
export interface DailyTodoRepository {
  findById(id: string): Promise<DailyTodo | undefined>;
  findByUserIdAndDate(userId: string, dayDate: string): Promise<DailyTodo[]>;
  countByUserIdAndDate(userId: string, dayDate: string): Promise<number>;
  save(item: DailyTodo): Promise<DailyTodo>;
  delete(id: string): Promise<void>;
}
